common/authorization/default_jwt_claim_mapper.go

```diff
@@ -159,11 +154,17 @@ func parseJWTWithAudience(tokenString string, keyProvider TokenKeyProvider, audi
 	if err != nil {
 		return nil, err
 	}
-
-	if claims, ok := token.Claims.(jwt.MapClaims); ok {
-		return claims, nil
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, serviceerror.NewPermissionDenied("invalid token with no claims", "")
+	}
+	if err := claims.Valid(); err != nil {
+		return nil, err
```

Comment:
will this be caught and returned as a permission denied error elsewhere?

Reply:
Yes, [here](https://github.com/temporalio/temporal/blob/3c427e6a2d5e7bcdda5868f379a26f6eb578871f/common/authorization/interceptor.go#L113).